nevergrad/functions/pyomo/core.py

```diff
@@ -0,0 +1,157 @@
+            if ranges[0].step in [-1, 1]:
+                params[params_name].set_integer_casting() #May consider using nested param
+        else:
+            raise NotImplementedError(f"Cannot handle range type {type(ranges[0])}")             
+    elif isinstance(domain, pyomo.FiniteSimpleRangeSet):
+        # Need to handle step size
+        params[params_name] = p.Choice([range(*r) for r in domain.ranges()]) #Assume the ranges do not overlapped
+    else:
+        raise NotImplementedError(f"Cannot handle domain type {type(domain)}")
+    return params
+
+
+def _make_pyomo_variable_to_parametrization(model_component : pyomo.Var, params : dict):
+    # https://pyomo.readthedocs.io/en/stable/pyomo_modeling_components/Sets.html
+    # Refer to the implementation in pyomo/core/base/var.py
+    # To further improve the readability function, we should find out how to represent {None: ng.p.Scalar(), 1: ng.p.Scalar()} in ng.p.Dict
+    # We do not adopt nested parameterization, which will require type information between string and int.
+    # Such conversion has to be done in _pyomo_obj_function_wrapper and _pyomo_constraint_wrapper, which slows down optimization.
+    if not (isinstance(model_component, pyomo.base.var.IndexedVar) or isinstance(model_component, pyomo.base.var.SimpleVar)):
+        raise NotImplementedError # Normally, Pyomo will create a set for the indices used by a variable
+
+    for k, v in model_component._data.items():
+        if isinstance(v, pyomo.base.var._GeneralVarData):
+            if v.is_fixed():
+                raise NotImplementedError
+            if k is None:
+                params_name = str(model_component.name)
+            else:
+                params_name = f"{model_component.name}[{_convert_to_ng_name(k)}]"
+            if isinstance(v.domain, pyomo.RangeSet):
+                params = _make_pyomo_range_set_to_parametrization(v.domain, params, params_name)
+            elif isinstance(v.domain, pyomo.Set) and v.domain.isfinite():
+                if v.domain.isordered():
+                    params[params_name] = p.Choice(list(v.domain.ordered_data()))
+                else:
+                    params[params_name] = p.Choice(list(v.domain.data()))
+            else:
+                raise NotImplementedError(f"Cannot handle domain type {type(v.domain)}")
+        else:
+            raise NotImplementedError(f"Cannot handle variable type {type(v)}")
+
+    return params
+
+
+class Pyomo(base.ExperimentFunction):
+    """Function calling Pyomo model
+
+    Parameters
+    ----------
+    name: str
+        problem name
+    model: pyomo.environ.model
+        Pyomo model
+
+    Returns
+    -------
+    float
+        the fitness
+
+    Notes
+    -----
+    - You will require an Pyomo installation (with pip: "pip install pyomo")
+    - Any changes on the model externally can lead to unexpected behaviours.
+    """
+
+    def __init__(self, name : str, model : pyomo.Model) -> None:
```

Comment:
i'm still unconvinced by the name, I'm quite confident (from experience.... :D ) that most people would just bypass this by writting `name="dummy"` every time and forget about it.

How about setting name as an optional parameter
```suggestion
    def __init__(self, model : pyomo.Model, name : tp.Optional[str] = None) -> None:
```
And either using the provided name if available, or deriving a string if not?

In any case, I would:
- make it the second argument and not the first, since the most important aspect is the model
- clarify the description to explain that it's only to "tag" the experiment in some way, and users can choose anything (it's basically not used except in the benchmarks)

Reply:
I have pushed new code to derive an identifier by joining the names of the objective functions, variable names, and constraint names.